### PR TITLE
Use native attributes

### PR DESCRIPTION
**Pull request by cpcallen**

### CODE REVIEW

File: server/interpreter.js [hunk 1]
```diff
@@ -3917,24 +3834,32 @@ Interpreter.prototype['stepUnaryExpression'] = function(stack, state, node) {
...
-            (name === 'length' && obj instanceof this.Array)) {
-          this.throwException(this.TYPE_ERROR, "Cannot delete property '" +
-                              name + "' of '" + obj + "'");
+        try {
+          value = delete obj.properties[key];
+        } catch (e) {
+          this.throwNativeException(e);
+        }
+      } else if (obj instanceof Interpreter.Sentinel) {
+        // Whoops; this should have been caught by acorn (because strict).
```
Comment (NeilFraser) [inline server/interpreter.js]: acorn -> Acorn

File: server/interpreter.js [hunk 2]
```diff
@@ -3917,24 +3834,32 @@ Interpreter.prototype['stepUnaryExpression'] = function(stack, state, node) {
...
+        throw Error("Uncaught illegal deletion of unqualified identifier");
```
Comment (NeilFraser) [inline server/interpreter.js]: Single quotes.

File: server/tests/db/test_01_es5.js [hunk 3]
```diff
@@ -1132,6 +1136,25 @@ tests.deleteProp = function() {
...
+};
+
+// This "actually" tries to delete the non-configurable own .length
+// property from the auto-boxed String instance created by step 4a of
+// algorithm in §11.4.1 of the ES 5.1 spec.  We have to use a string
+// here, because only String instances have own properties (and yes:
+// they are all non-configurable, so delete *always* fails).
+tests.deleteOwnFromPrimitive = function() {
+  try {
+    delete "hello".length;
```
Comment (NeilFraser) [inline server/tests/db/test_01_es5.js]: Single quotes.

Comment (cpcallen) [reply server/tests/db/test_01_es5.js]: Not that the tests are remotely consistent about that otherwise…

**Pull request merged**